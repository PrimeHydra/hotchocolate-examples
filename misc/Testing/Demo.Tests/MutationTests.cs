namespace Demo.Tests;

public class MutationTests : QueryTestBase
{
    [Fact]
    public async Task AddSparkyAndReadHimBack()
    {
        await this.ExecuteQueryAsync(
            @"
mutation {
  addPuppy(input: { name: ""Sparky"", breed: ""Beagle"" }) {
    id
  }
}
                ");

        var result = await this.ExecuteQueryAsync(
            @"
{
  puppy(id: ""Id_1"") {
    name
    breed
  }
}

                   
                ");

        result.MatchSnapshot();
    }

    [Fact]
    public async Task AddDaisyAndReadHerBack()
    {
        await this.ExecuteQueryAsync(
            @"
mutation {
  addPuppy(input: { name: ""Daisy"", breed: ""Golden Doodle"" }) {
    id
  }
}
                ");

        var result = await this.ExecuteQueryAsync(
            @"
{
  puppy(id: ""Id_1"") {
    name
    breed
  }
}

                   
                ");

        result.MatchSnapshot();
    }

}