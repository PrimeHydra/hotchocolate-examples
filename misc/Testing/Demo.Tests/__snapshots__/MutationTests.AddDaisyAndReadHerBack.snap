{
  "data": {
    "puppy": {
      "name": "Daisy",
      "breed": "Golden Doodle"
    }
  }
}
